src/sample/View.java

```diff
@@ -189,16 +185,15 @@ else if(treasure.getX() == currentColumnIndex && treasure.getY() == currentRowIn
         }
     }
 
-    // sets up the buttons and adds them to the grid pane
-    // The 3 consumers sent to this method should NOT TAKE AN INPUT.  We had to put the object to get around being
-    // forced to send something to the method.
     void setupButtons(Consumer<Object> nextFunction, Consumer<Object> resetFunction, Consumer<Object> autoPlayFunction,
```

Comment:
This whole thing is becoming ugly, I think we need to rethink this pattern

Reply:
I agree this one may take some rethinking and plus the button with args is not in this one